methods/lcarun.ts

```diff
@@ -2,28 +2,106 @@ import knex from 'knex';
 
 import { Lci } from 'models/schema';
 import { RunParams } from './lca.model';
-// import { listenerCount } from 'cluster';
 
+// NonrenewableUnitEnergyContent were used to change units for nonrenewable group.
+// Units change is not needed for other groups
+interface NonrenewableUnitEnergyContent {
+  brownCoal: number;
+  hardCoal: number;
+  crudeOil: number;
+  mineGas: number;
+  naturalGas: number;
+  Uranium: number;
+  [key: string]: number;
+}
+// unitEnergyContent refers to the energy content per kg of the flow.
+// For example, the weight of brownCoal can be queried from table lci_input; its energy content is 9.9 MJ/kg.
+// The total energy content of brownCoal can be calculate by multiplying the weight by energy content per kg.
+const unitEnergyContent: NonrenewableUnitEnergyContent = { brownCoal: 9.9, hardCoal: 19.1, crudeOil: 45.8,
+                                                           mineGas: 32.45, naturalGas: 32.12, Uranium: 560000};
 const lcarun = async (params: RunParams, db: knex) => {
-  const rows: Lci[] = await db
-    .table('lci_input')
-    .where({ lci_group: 'renewable' });
-
-  // const bigNum = Math.max(...rows.map(r => r.number));
-  let sum = 0;
-  for (let i = 0; i < rows.length; i++) {
-      sum = sum + processRow(rows[i], params);
+  const consumptions = await processInput(params, db);
+  const pollutions = await processOutput(params, db);
+
+  return { success: true, params, consumptions, pollutions };
+};
+
+const processInput = async (params: RunParams, db: knex) => {
```

Comment:
i would change this to "calculateEnergyConsumption" or "processEnergyConsumption", something like that